Main.py

```diff
@@ -1,62 +1,249 @@
+    #sql query to get rate inserting currency into where clause
+        res=cur.execute("SELECT rate FROM rates WHERE currency = ?", (currency,))
+        x = res.fetchall()
+        #x is a list of tuples, need to separate out rate as float
+        conversionRate = x[0][0]
+        return conversionRate
+    
+    #function to check if user input currency is in fact in the db
+    def check_for_currency(currency, cur)->bool:
+        #sql query to return a boolean
+        res=cur.execute("SELECT rate FROM rates WHERE currency = ?", (currency,))
+        queryBool = res.fetchone() is not None
+
+        #this if statement handles returning the rate as float
+        if queryBool is True:
+            print (f'Currency is {currency}')
+            return float(APIService.currencyRate(currency,cur))
+
+        #if the currency doesn't exist it gives the user an error
+        else:
+            print ('Currency does not exist.')
+
+
+"""
+Mock Data Handler Class in case API not working
+"""
+class MockDataService(AbstractCurrencyService):
+    #will need to overwrite csv every time
+    def get_currency_data():
+        print("API not connected-out of date exchange rates in use")
+       
+    def getCurrency():
+        print("Please type in the abbreviation of the currency in which the donation will be received")
+        donorCurrency= input()
+        donCurrCaps = donorCurrency.upper()
+        return donCurrCaps
+    
+    def currencyRate(dc, cur):
+    #sql query to get rate inserting currency into where clause
+        res=cur.execute("SELECT rate FROM rates WHERE currency = ?", (dc,))
+        x = res.fetchall()
+
+        #x is a list of tuples, need to separate out rate as float
+        conversionRate = x[0][0]
+        return conversionRate
+    
+    def check_for_currency(dc, cur)->bool:
+        #sql query to return a boolean
+        res=cur.execute("SELECT rate FROM rates WHERE currency = ?", (dc,))
+        queryBool = res.fetchone() is not None
+
+        #this if statement handles returning the rate as float
+        if queryBool is True:
+            print (f'Currency is {dc}')
+            return float(APIService.currencyRate(dc,cur))
+
+        #if the currency doesn't exist it gives the user an error
+        else:
+            print ('Currency does not exist.')
+
+
+    
+"""
+Main
+"""
+#1)Check if converting currency
+# 2)check if update datetime align
+#  A)if yes, call api and get response
+#    1)if api response is 200, update the db and csv and use API service
+#    2)if api response is NOT 200 use mock data service (which will just provide un-updated db with warning)
+#  B)if no, do not call api and proceed with program
+
 
 #gofundme fees, 2.9% + $.30usd
 feePercentage=.029
 flatFeeAddOn=.3 #will need to convert this when converting currency
-#amountIntended
-#floatIntended
+
 print("Converting between any currencies? Y or N")
 converting=input()
 #if yes
 if(converting == "Y" or converting =="y"):
```

Comment:
For use cases like these you can use `.lower()` or `.upper()` to convert the input into a standard format for comparision.